src/scripts/main.js

```diff
@@ -1,3 +1,339 @@
+                </p>
+              </div>
+             `;
+      break;
+    case 'error':
+      notification = `<div 
+                data-qa="notification" 
+                class="notification ${type}"
+              >
+                <p class="title">
+                  Invalid data.
+                  Сheck the following:
+                  * Name value has not less than 4 letters
+                  * Age is not less than 18 or more than 90
+                </p>
+              </div>
+             `;
+      break;
+  }
+  root.insertAdjacentHTML('afterbegin', notification);
+
+  setTimeout(() => {
+    const message = root.querySelector('.notification');
+
+    message.remove();
+  }, 3000);
+};
+
+const makeSalaryValue = num => {
+  const correctNum = num.toFixed(3);
+
+  return `$${correctNum.toString().replace('.', ',')}`;
+};
+
+table.insertAdjacentHTML('afterend', formEl);
+
+const form = document.querySelector('form');
+
+form.addEventListener('submit', e => {
+  e.preventDefault();
+
+  const formData = new FormData(form);
+  const tableData = Object.fromEntries(formData.entries());
+
+  if (!formValidator(tableData)) {
+    addNotification('error');
+
+    return;
+  }
+
+  const newRow = `
+    <tr>
+      ${Object.keys(tableData).map(key => {
+    if (key === 'salary') {
+      return `<td>${makeSalaryValue(+tableData[key])}</td>`;
+    }
+
+    return `<td>${tableData[key]}</td>`;
+  }).join('')}
+    </tr>
+  `;
+
+  tBody.insertAdjacentHTML('beforeend', newRow);
+  addNotification('success');
+  form.reset();
+});
+
+// editing of table cells;
+
+const getCellType = (value) => {
+  let cellType;
+
+  switch (true) {
+    case value.includes('$'):
+      cellType = 'salary';
+      break;
+
+    case offices.includes(value):
+      cellType = 'office';
+      break;
+
+    case !isNaN(+value):
+      cellType = 'age';
+      break;
+
+    case typeof value === 'string':
+      cellType = 'str';
+      break;
+  }
+
+  return cellType;
+};
+
+const validateCell = (cellType, newValue) => {
+  switch (cellType) {
+    case 'str':
+      if (newValue.length > 2) {
+        return newValue;
+      }
+      break;
+
+    case 'office':
+      if (offices.includes(newValue)) {
+        return newValue;
+      }
+      break;
+
+    case 'age':
+      if (newValue >= 18 && newValue <= 90) {
+        return newValue;
+      }
+      break;
+
+    case 'salary':
+      if (!isNaN(parseFloat(newValue.replace('$', '')))) {
+        return `$${parseFloat(
+          newValue
+            .replace('$', '')
+            .replace(',', '.')
+            .replace('-', ''))
+          .toFixed(3)
+        }`;
+      }
+  }
+};
+
+tBody.addEventListener('dblclick', bodyEvent => {
+  const initialValue = bodyEvent.target.textContent;
+
+  bodyEvent.target.innerHTML = `<input 
+                                  type="text" 
+                                  class="cell-input"  
+                                  value="${initialValue}">`;
```

Comment:
it can be better for code style

```suggestion
  bodyEvent.target.innerHTML = `
                              <input 
                                  type="text" 
                                  class="cell-input"  
                                  value="${initialValue}"
                               >
 `;
```

Reply:
then i have nodeText (spaces) afte and before node